### PR TITLE
MBS-10116: Convert the recording index page to React

### PR DESCRIPTION
**Pull request by reosarevok**
https://tickets.metabrainz.org/browse/MBS-10116

Please do review the flow types specifically, since those still confuse me.

### CODE REVIEW

File: lib/MusicBrainz/Server/Entity/Track.pm [hunk 1]
```diff
@@ -68,6 +68,7 @@ around TO_JSON => sub {
         %{ $self->$orig },
         isDataTrack     => boolean_to_json($self->is_data_track),
         length          => $self->length,
+        medium          => $self->medium->TO_JSON;
```
Comment (mwiencek) [inline lib/MusicBrainz/Server/Entity/Track.pm]: (Haven't tested the PR yet, but: syntax error here with `;` at the end.)

File: root/recording/RecordingIndex.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  +numberOfRevisions: number,
+  +pager: PagerT,
+  +recording: RecordingT,
+  +tracks: $ReadOnlyArray<$ReadOnlyArray<{...TrackT, +medium: MediumT}>>,
+|};
+
+const RecordingAppearancesTable = ({
+  tracks,
+}: {
+  tracks: $ReadOnlyArray<$ReadOnlyArray<{...TrackT, +medium: MediumT}>>,
```
Comment (mwiencek) [inline root/recording/RecordingIndex.js]: You could use `$ElementType<Props, 'tracks'>` there to simplify things a bit.

**Comment by mwiencek** [comment]
Can we also remove root/recording/index.tt now?

**Comment by reosarevok** [comment]
Feel free to change loopParity to work that way, if you think it's better! :)

**Comment by mwiencek** [comment]
Done so in https://github.com/metabrainz/musicbrainz-server/commit/e8cb66244bd3869f23e24bc0bdefcf798af27b74

**Pull request merged**